commit f5499a001d9cbde6a672f9d393d9c80c63745f00 (from 0ccd5c802b94fc9206e4ce27a9fbd4bebc66a771)
Merge: 0ccd5c802b94 1cf510d1decf
Author: bors <bors@rust-lang.org>
Date:   Wed Aug 24 23:05:47 2016 -0700

    Auto merge of #35814 - alexcrichton:armv7-no-neon, r=brson
    
    rustc: Don't enable NEON by default on armv7 Linux
    
    One of the primary platforms for the `armv7-unknown-linux-gnueabihf` target,
    Linux distributions, do not enable NEON extensions by default. This PR disables
    that feature by defualt but enables the `d16` feature which enables VFP3D16 that
    distributions do enable.
    
    Closes #35590

diff --git a/src/librustc_back/target/armv7_unknown_linux_gnueabihf.rs b/src/librustc_back/target/armv7_unknown_linux_gnueabihf.rs
index 52269f0cd4a0..7e0306a03e2f 100644
--- a/src/librustc_back/target/armv7_unknown_linux_gnueabihf.rs
+++ b/src/librustc_back/target/armv7_unknown_linux_gnueabihf.rs
@@ -23,8 +23,9 @@ pub fn target() -> TargetResult {
         target_vendor: "unknown".to_string(),
 
         options: TargetOptions {
-            features: "+v7,+vfp3,+neon".to_string(),
-            cpu: "cortex-a8".to_string(),
+            // Info about features at https://wiki.debian.org/ArmHardFloatPort
+            features: "+v7,+vfp3,+d16,+thumb2".to_string(),
+            cpu: "generic".to_string(),
             max_atomic_width: 64,
             .. base
         }
